### ISSUE TITLE
NUMA Pinning

### ISSUE THREAD

**Issue by mrks** [hyrise/hyrise#947]
If we are on a NUMA system and we don't set anything, the Linux kernel wildly migrates our process. No idea why, and it doesn't really make sense.

We should do some pinning for all cases where we don't use the NumaPlacementManager.

**Comment by mrks** [comment]
The server profits immensely from being run on a single NUMA node. We should consider pinning either the server's main process or all sessions on a single node.

Benchmark:
```
echo 'SELECT * FROM t;' > test
echo 'CREATE TABLE t (a INT)' | psql -h localhost -p 1241
pgbench -h localhost -p 1241 -f test -n -c 100 -j 100 -T 10
```

Results:
| NUMA Binding                      	| Q/s        	|
|-----------------------------------	|------------	|
| numactl -N 0 ./hyriseServer       	| 159797     	|
| numactl -N 0,1 ./hyriseServer     	| 51153      	|
| numactl -N 0,1,2 ./hyriseServer   	| 35570      	|
| numactl -N 0,1,2,3 ./hyriseServer 	| 65839 (!?) 	|

Also pinging @Bouncner and @Bensk1, who I had a discussion about the server throughput with.

**Comment by tobodner** [comment]
I suspect that NUMA node 0 makes the difference here, since it owns the network device. Have you also benchmarked just node 1, 2, 3? It would be interesting for validation purposes.

**Comment by mrks** [comment]
Good point. I added some more numbers above. Interestingly which NUMA node is used does make a difference and 0 is particularly *bad*.

Note that these are loopback benchmarks. When looking at this, we should also run benchmarks through the network.
